client/web/compose/src/views/Admin/Pages/Builder.vue

```diff
@@ -482,75 +534,44 @@ export default {
-          previousTab = this.editor.block.options.tabs[passedBlock.tabIndex]
-        }
-
-        this.editor.block.options.tabs.splice(passedBlock.tabIndex, 1, tab)
-        this.$root.$emit('builder-createRequestFulfilled', { previousTab, block })
-      } else {
-        this.editor.block.meta.tempID = this.editor.block.meta.tempID || this.generateUID()
-        const block = compose.PageBlockMaker(this.editor.block)
-        this.page.blocks = this.blocks
+    updateBlocks (block = this.editor.block) {
+      block = compose.PageBlockMaker(block)
+      this.page.blocks = this.blocks
 
-        if (this.editor.index !== undefined) {
+      if (this.editor.index !== undefined) {
+        if (this.editor.block.kind !== block.kind) {
+          this.page.blocks.push(block)
+          const tab = {
+            blockID: fetchID(block),
+            title: block.title,
+          }
+          this.$root.$emit('builder-createRequestFulfilled', { tab })
+        } else {
           this.page.blocks.splice(this.editor.index, 1, block)
           this.unsavedBlocks.add(this.editor.index)
-        } else {
-          this.page.blocks.push(block)
-          this.unsavedBlocks.add(this.page.blocks.length - 1)
-        }
-
-        if (this.editor.block.kind === 'Tabs') {
-          this.makeTab(block)
-          block.options.tabs.forEach((tab) => {
-            this.findBlock(tab.block).meta.tabbed = true
-          })
         }
-        this.editor = undefined
+      } else {
+        this.page.blocks.push(block)
+        this.unsavedBlocks.add(this.page.blocks.length - 1)
       }
-    },
-
-    cloneBlock (index) {
-      this.appendBlock({ ...this.blocks[index] }, this.$t('notification:page.cloneSuccess'))
-    },
-
-    makeTab (block) {
-      if (!block.options.tabs.length) return
-      block.options.tabs.forEach((tab, index) => {
-        if (tab.block !== null) {
-          this.createTab(tab, index, block)
-        }
-      })
-      this.cleanUpTabs(block)
-    },
 
-    createTab (tab, tabIndex, block) {
-      const blockToTab = this.findBlock(tab.block)
-      const newTab = {
-        block: tab.block,
-        title: tab.title || blockToTab.title,
+      if (block.kind === 'Tabs') {
+        block.options.tabs = block.options.tabs.filter(t => t.blockID !== null)
+        block.options.tabs.forEach((tab) => {
+          this.page.blocks.find(b => fetchID(b) === tab.blockID).meta.tabbed = true
+        })
       }
-      this.updateTab(newTab, tabIndex, block)
-    },
 
-    updateTab (newTab, tabIndex, block) {
-      if (this.findBlock(newTab.block).kind === 'Tabs') return
-      block.options.tabs[tabIndex] = newTab
+      if (this.editor.block.kind === block.kind) {
+        this.editor = undefined
+      }
     },
 
-    cleanUpTabs (block) {
-      block.options.tabs = block.options.tabs.filter(t => t.block !== null)
+    cloneBlock (index) {
+      const block = compose.PageBlockMaker({ ...this.blocks[index] })
```

Comment:
Instead of doing it in three lines, just do { ...this.blocks[index], blockID: NoID, meta: { tempID: '' }, so you can reduce this to one line

```
this.appendBlock(compose.PageBlockMaker({
   ...this.blocks[index],
   blockID: NoID,
   meta: { tempID: '' }
})
```

you could also add a clone function to the block class itself so you don't have to care about such things, which would be the better approach